esi/lease/v1/_common.py

```diff
@@ -0,0 +1,41 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+# borrowed from openstacksdk
+class ListMixin:
+    @classmethod
+    def list(cls, session, details=False, **params):
+        """This method is a generator which yields resource objects.
+
+        This resource object list generator handles pagination and takes query
+        params for response filtering.
+
+        :param session: The session to use for making this request.
+        :type session: :class:`~keystoneauth1.adapter.Adapter`
+        :param bool details: Whether to return detailed node records
+        :param dict params: These keyword arguments are passed through the
+            :meth:`~openstack.resource.QueryParameter._transpose` method
+            to find if any of them match expected query parameters to be
+            sent in the *params* argument to
+            :meth:`~keystoneauth1.adapter.Adapter.get`.
+
+        :return: A generator of :class:`openstack.resource.Resource` objects.
+        :raises: :exc:`~openstack.exceptions.InvalidResourceQuery` if query
+                 contains invalid params.
+        """
+        base_path = cls.base_path
+        if details:
+            base_path += '/detail'
```

Comment:
I don't think ESI-Leap actually has a 'detail' option; it's limited to the python-esileapclient, which filters out fields itself. So I don't think this detail flag actually does anything - maybe I'm misunderstanding?

Reply:
I think you are right. Detail flag is inherited from openstacksdk resource class. I'll remove it.